src/Forms/GridField/GridField.php

```diff
@@ -1117,11 +1116,7 @@ public function getColumnMetadata($column)
      */
     public function getColumnCount()
     {
-        if (!$this->columnDispatch) {
-            $this->buildColumnDispatch();
-        }
-
-        return count($this->columnDispatch ?? []);
+        return count($this->getColumns());
```

Comment:
`columnDispatch` is based on `getColumnsHandled()` (what columns can we _theoretically_ have?), but we should be using `getColumns()` instead which is based on `augmentColumns()` (what columns do we _actually_ have?)